### PR TITLE
BS-14941 create a service to store runtime data on tenant

### PR DESCRIPTION
**Pull request by baptistemesta**
this is for storing resources of the tenant like properties files, bdm things and so on.
It uses the table tenant_resource with the sequenceid 10501.
In the meantime ResourcesService was renamee as long with its package to ProcessResourcesService.

### CODE REVIEW

File: bonita-integration-tests/bonita-integration-tests-local/src/test/java/org/bonitasoft/engine/resources/TenantResourcesServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import org.bonitasoft.engine.bpm.CommonBPMServicesTest;
+import org.bonitasoft.engine.transaction.TransactionService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Nicolas Chabanoles
+ * @author Celine Souchet
```
Comment (educhastenier) [inline bonita-integration-tests/bonita-integration-tests-local/src/test/java/org/bonitasoft/engine/resources/TenantResourcesServiceTest.java]: Sure?

File: bonita-integration-tests/bonita-integration-tests-local/src/test/java/org/bonitasoft/engine/resources/TenantResourcesServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+public class TenantResourcesServiceTest extends CommonBPMServicesTest {
+
+    private static TenantResourcesService tenantResourcesService;
+
+    private static TransactionService transactionService;
+
+
+    @Before
+    public void before(){
```
Comment (educhastenier) [inline bonita-integration-tests/bonita-integration-tests-local/src/test/java/org/bonitasoft/engine/resources/TenantResourcesServiceTest.java]: format user Eclipse Formatter please

File: bonita-integration-tests/bonita-integration-tests-local/src/test/java/org/bonitasoft/engine/resources/TenantResourcesServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        tenantResourcesService = getTenantAccessor().getTenantResourcesService();
+        transactionService = getTransactionService();
+    }
+
+    @After
+    public void after() throws Exception {
+        transactionService.executeInTransaction(new Callable<Object>() {
+            @Override
+            public Object call() throws Exception {
```
Comment (educhastenier) [inline bonita-integration-tests/bonita-integration-tests-local/src/test/java/org/bonitasoft/engine/resources/TenantResourcesServiceTest.java]: Cannot you return Void?

File: bpm/bonita-sql/src/main/resources/sql/h2/createTables.sql [hunk 4]
```diff
@@ -1034,3 +1034,14 @@ CREATE TABLE bar_resource (
 );
 CREATE INDEX idx_bar_resource ON bar_resource (tenantId, process_id, type, name);
 
+CREATE TABLE tenant_resource (
+  tenantId BIGINT NOT NULL,
+  id BIGINT NOT NULL,
+  name VARCHAR(255) NOT NULL,
+  type VARCHAR(16) NOT NULL,
+  content LONGBLOB NOT NULL,
+  UNIQUE (tenantId, name, type),
```
Comment (educhastenier) [inline bpm/bonita-sql/src/main/resources/sql/h2/createTables.sql]: Please extract the constraints in a separate line and give it a Unique name

File: bpm/bonita-sql/src/main/resources/sql/h2/createTables.sql [hunk 5]
```diff
@@ -1034,3 +1034,14 @@ CREATE TABLE bar_resource (
...
+  PRIMARY KEY (tenantId, id)
```
Comment (educhastenier) [inline bpm/bonita-sql/src/main/resources/sql/h2/createTables.sql]: also for primary key

File: bpm/bonita-home/src/main/resources/bonita-platform-community.xml [hunk 6]
```diff
@@ -623,13 +623,22 @@
...
         <constructor-arg name="sequenceId" value="10500" />
         <constructor-arg name="rangeSize" value="${bonita.platform.sequence.10500:${bonita.platform.sequence.defaultRangeSize}}" />
     </bean>
     <bean class="org.bonitasoft.engine.sequence.SequenceMapping">
+        <constructor-arg name="classNames">
+            <set>
+                <value>org.bonitasoft.engine.resources.STenantResource</value>
+            </set>
+        </constructor-arg>
+        <constructor-arg name="sequenceId" value="10501" />
```
Comment (educhastenier) [inline bpm/bonita-home/src/main/resources/bonita-platform-community.xml]: Please add sequence mapping in the TXT file as well

File: bpm/bonita-sql/src/main/resources/sql/mysql/createTables.sql [hunk 7]
```diff
@@ -1042,3 +1042,14 @@ CREATE TABLE bar_resource (
   PRIMARY KEY (tenantId, id)
 ) ENGINE = INNODB;
 CREATE INDEX idx_bar_resource ON bar_resource (tenantId, process_id, type, name);
+
+CREATE TABLE tenant_resource (
+  tenantId BIGINT NOT NULL,
+  id BIGINT NOT NULL,
+  name VARCHAR(255) NOT NULL,
+  type VARCHAR(16) NOT NULL,
```
Comment (educhastenier) [inline bpm/bonita-sql/src/main/resources/sql/mysql/createTables.sql]: With 16, we are good for all future Enum values?

File: bonita-integration-tests/bonita-query-tests/src/test/java/org/bonitasoft/engine/resources/TenantResourceServiceQueriesTest.java [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+        //when
+        STenantResource myResource = repository.getTenantResource(TenantResourceType.BDM, "MyResource");
+        // //then
+        assertThat(myResource.getContent()).isEqualTo("The content".getBytes());
+        assertThat(myResource.getId()).isEqualTo(resource.getId());
+    }
+
+    @Test
+    public void getTenantResourceOfType_should_get_all_resource_of_type() {
```
Comment (educhastenier) [inline bonita-integration-tests/bonita-query-tests/src/test/java/org/bonitasoft/engine/resources/TenantResourceServiceQueriesTest.java]: Maybe you can add a resource of another type to make sure it is NOT retrieved...

File: bpm/bonita-sql/src/main/resources/sql/h2/createTables.sql (hunk 4 above)
Comment (baptistemesta) [inline bpm/bonita-sql/src/main/resources/sql/h2/createTables.sql]: ok

File: bonita-integration-tests/bonita-integration-tests-local/src/test/java/org/bonitasoft/engine/resources/TenantResourcesServiceTest.java (hunk 3 above)
Comment (baptistemesta) [inline bonita-integration-tests/bonita-integration-tests-local/src/test/java/org/bonitasoft/engine/resources/TenantResourcesServiceTest.java]: I could... but i think it's ok like that

File: bpm/bonita-sql/src/main/resources/sql/h2/createTables.sql (hunk 5 above)
Comment (baptistemesta) [inline bpm/bonita-sql/src/main/resources/sql/h2/createTables.sql]: we have that for foreign keys, not primary

File: bpm/bonita-home/src/main/resources/bonita-platform-community.xml (hunk 6 above)
Comment (baptistemesta) [inline bpm/bonita-home/src/main/resources/bonita-platform-community.xml]: ok

File: bonita-integration-tests/bonita-query-tests/src/test/java/org/bonitasoft/engine/resources/TenantResourceServiceQueriesTest.java (hunk 8 above)
Comment (baptistemesta) [inline bonita-integration-tests/bonita-query-tests/src/test/java/org/bonitasoft/engine/resources/TenantResourceServiceQueriesTest.java]: I think I prefer to add this test when we have a second resource to persist with the tenant resources service> I don't want to add a fake resource or decide in advance what there will be there

File: bpm/bonita-sql/src/main/resources/sql/mysql/createTables.sql (hunk 7 above)
Comment (baptistemesta) [inline bpm/bonita-sql/src/main/resources/sql/mysql/createTables.sql]: I think yes, as long as we choose the name we are ok

**Pull request merged**